arena.go

```diff
@@ -0,0 +1,66 @@
+package bigqueue
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/grandecola/mmap"
+)
+
+const (
+	cFilePerm = 0744
+)
+
+// Arena is an abstraction for a memory mapped file of a given size
+type Arena struct {
+	*mmap.Mmap
+	size int
+	file string
+}
+
+// NewArena returns pointer to an Arena. It takes a file location and mmaps it.
+// If file location does not exist, it creates file of given size.
+func NewArena(file string, size int) (*Arena, error) {
+	fd, err := openOrCreateFile(file, size)
+	if err != nil {
+		return nil, err
+	}
+
+	m, err := mmap.NewSharedFileMmap(fd, 0, size, syscall.PROT_READ|syscall.PROT_WRITE)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Arena{
+		Mmap: m,
+		size: size,
+		file: file,
+	}, nil
+}
+
+func openOrCreateFile(file string, size int) (*os.File, error) {
+	if _, err := os.Stat(file); err == nil {
+		// open file
+		fd, err := os.OpenFile(file, os.O_RDWR, cFilePerm)
```

Comment:
We should open existing file with custom permissions. While creating file, we should use these permissions. Just use os.Open() here.

Reply:
We need to open file for reading and writing. Simply using `os.Open` will only open the file for reading.